shell/platform/android/test/io/flutter/plugin/platform/PlatformPluginTest.java

```diff
@@ -416,93 +507,59 @@ public void setSystemUiMode() {
-                PlatformChannel.SystemUiOverlay.BOTTOM_OVERLAYS)));
-
-    verify(fakeDecorView).setSystemUiVisibility(View.SYSTEM_UI_FLAG_FULLSCREEN);
-    verify(fakeDecorView).setSystemUiVisibility(View.SYSTEM_UI_FLAG_HIDE_NAVIGATION);
-    verify(fakeDecorView)
-        .setSystemUiVisibility(
-            View.SYSTEM_UI_FLAG_LAYOUT_STABLE
-                | View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
-                | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN);
-
-    verify(fakeDecorView).setSystemUiVisibility(fakeSetFlags &= ~View.SYSTEM_UI_FLAG_FULLSCREEN);
-    verify(fakeDecorView)
-        .setSystemUiVisibility(fakeSetFlags &= ~View.SYSTEM_UI_FLAG_HIDE_NAVIGATION);
-  }
-
-  @TargetApi(30)
-  @Test
-  public void showSystemOverlays() {
-    View fakeDecorView = mock(View.class);
-    Window fakeWindow = mock(Window.class);
-    when(fakeWindow.getDecorView()).thenReturn(fakeDecorView);
-    Activity fakeActivity = mock(Activity.class);
-    when(fakeActivity.getWindow()).thenReturn(fakeWindow);
-    PlatformChannel fakePlatformChannel = mock(PlatformChannel.class);
-    PlatformPlugin platformPlugin = new PlatformPlugin(fakeActivity, fakePlatformChannel);
+    // WindowInsetsControllerCompat windowInsetsControllerCompat = new
+    // WindowInsetsControllerCompat(fakeWindow, testView);
     WindowInsetsController fakeWindowInsetsController = mock(WindowInsetsController.class);
     when(fakeWindow.getInsetsController()).thenReturn(fakeWindowInsetsController);
 
-    platformPlugin.mPlatformMessageHandler.showSystemOverlays(
-        new ArrayList<PlatformChannel.SystemUiOverlay>());
-
-    verify(fakeWindowInsetsController)
-        .setSystemBarsBehavior(WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE);
-    verify(fakeWindowInsetsController).hide(WindowInsetsCompat.Type.systemBars());
-
-    platformPlugin.mPlatformMessageHandler.showSystemOverlays(
-        new ArrayList<PlatformChannel.SystemUiOverlay>(
-            Arrays.asList(
-                PlatformChannel.SystemUiOverlay.TOP_OVERLAYS,
-                PlatformChannel.SystemUiOverlay.BOTTOM_OVERLAYS)));
-
-    verify(fakeWindowInsetsController).show(WindowInsetsCompat.Type.statusBars());
-    verify(fakeWindowInsetsController).show(WindowInsetsCompat.Type.navigationBars());
+    platformPlugin.mPlatformMessageHandler.setSystemUiChangeListener();
+
+    /* All of the attemps below resulted in the call: fakePlatformCannel.systemChromeCanged(true) */
+
+    // (1) Attempt with using listnener directly (I exposed it by adding a getInsetsListener method)
+    WindowInsets.Builder builder = new WindowInsets.Builder();
+    builder.setInsets(WindowInsetsCompat.Type.systemBars(), Insets.of(0, 200, 0, 200));
+    WindowInsets fullScreenInsets = builder.build();
+
+    platformPlugin
+        .getInsetsListener()
+        .onApplyWindowInsets(testView, WindowInsetsCompat.toWindowInsetsCompat(fullScreenInsets));
```

Comment:
I am not sure I follow this correctly, but it looks like you are setting the status bars to be visible, which means `fakePlatformCannel.systemChromeCanged(true)` is correct?

Reply:
No, it's supposed to return false if the system bars are visible. So the case below with edge to edge should return true, but it's returning true no matter the case which is the problem

Reply:
It may not be working then. Have you tried this with a sample app to verify it works as expected? The test may be validly showing it is not working.

Reply:
My concern is validly mocking a `WindowInsetsController`/`WindowInsetsControllerCompat` because the issue is that calling `show` with these to make the system bars visible is not working the way I have it (I checked with `isVisible(...)`).  This is why it always calls `fakePlatformCannel.systemChromeChanged(true)` right now. I wanted to ask about this because I assumed this wasn't test for a similar reason before...none of the tests for the `PlatformPlugin` concern the actual visibility of overlays or insets, only that the correct methods were called/flags were set.